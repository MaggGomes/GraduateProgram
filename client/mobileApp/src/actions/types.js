export const FETCH_EVENTS = 'FETCH_EVENTS';
export const FETCH_EVENTS_ERROR = 'FETCH_EVENTS_ERROR';
export const FETCH_DAY = 'FETCH_DAY';
export const FETCH_DAY_ERROR = 'FETCH_DAY';
export const FETCH_GALLERY = 'FETCH_GALLERY';
export const FETCH_GALLERY_ERROR = 'FETCH_GALLERY';
