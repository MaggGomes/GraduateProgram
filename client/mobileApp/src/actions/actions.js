import {FETCH_EVENTS} from './types';
import axios from 'axios';

export const fetchEvents = () => dispatch =>{
    axios.get('https://graduates-mindera.herokuapp.com/events')
        .then((res) => dispatch({
            type: FETCH_EVENTS,
            payload: res.data
        }))
        .catch((error) =>{

        });
};