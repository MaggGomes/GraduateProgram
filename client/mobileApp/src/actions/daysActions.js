import {FETCH_DAY, FETCH_DAY_ERROR} from './types';
import axios from 'axios';

export const fetchDay = (id) => dispatch =>{
    axios.get('https://graduates-mindera.herokuapp.com/day/'+id)
        .then((res) => dispatch({
            type: FETCH_DAY,
            payload: res.data
        }))
        .catch((error) => dispatch({
            type: FETCH_DAY_ERROR,
            payload: error
        }));
};