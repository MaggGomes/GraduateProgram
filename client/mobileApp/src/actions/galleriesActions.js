import {FETCH_GALLERY} from './types';
import axios from 'axios';

export const fetchGallery = (id) => dispatch =>{
    axios.get('https://graduates-mindera.herokuapp.com/gallery/'+id)
        .then((res) => dispatch({
            type: FETCH_GALLERY,
            payload: res.data
        }))
        .catch((error) => dispatch({
            payload: error
        }));
};