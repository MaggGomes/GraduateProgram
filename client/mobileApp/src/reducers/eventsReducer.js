import {FETCH_EVENTS} from '../actions/types'

const initialState = {
    data: [],
    loading: false,
    error: null
};

export default function(state=initialState, action){
    switch(action.type){
        case FETCH_EVENTS:
            return {
                ...state,
                loading: false,
                data: action.payload
            };
        default:
            return state;
    }
}