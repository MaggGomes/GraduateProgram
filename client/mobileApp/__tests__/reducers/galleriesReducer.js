import React from 'react';
import eventsReducer from '../../src/reducers/eventsReducer';
import {FETCH_GALLERY, FETCH_GALLERY_ERROR} from '../../src/actions/types';
import {fetchGallery} from "../../src/actions/galleriesActions";

describe('galleriesReducer', () => {
    it('FETCH_GALLERY', () => {
        expect(eventsReducer(undefined, fetchGallery(1))).toMatchSnapshot();
    });
    it('FETCH_GALLERY_ERROR', () => {
        const action = { type: FETCH_GALLERY_ERROR };
        expect(eventsReducer(undefined, action)).toMatchSnapshot();
    });
});
