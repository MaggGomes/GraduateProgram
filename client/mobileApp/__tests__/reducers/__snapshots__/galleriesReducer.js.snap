// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`galleriesReducer FETCH_GALLERY 1`] = `
Object {
  "data": Array [],
  "error": null,
  "loading": true,
}
`;

exports[`galleriesReducer FETCH_GALLERY_ERROR 1`] = `
Object {
  "data": Array [],
  "error": null,
  "loading": true,
}
`;
